packages/overlay/src/Overlay.ts

```diff
@@ -516,6 +516,13 @@ export class Overlay extends ComputedOverlayBase {
             return;
         }
 
+        // Earlier this nextFrame() was called in ensureOnDom. Without waiting here, the longpress state will be `null` when the overlay is opened and the overlay will close immediately.
+        // I moved it here because this nextFrame messes up the tray inside a modal in Safari.
+        // So now we conditionally wait for the next frame only if the trigger interaction is longpress.
+        if (this.triggerInteraction === 'longpress') {
+            await nextFrame();
+        }
+
```

Comment:
```suggestion
// Only wait for next frame if `longpress` is the trigger.
// In Safari, awaiting nextFrame here causes layout issues
// when rendering trays inside modals, so we skip it otherwise.
if (this.triggerInteraction === 'longpress') {
    await nextFrame();
}
```